TemplateLint exception

So I've built v2025.1.2 of the plugin from source, some exception popped while working on a file (noticed it too late to find out what I was doing though)

```log
ExternalToolPass: 

com.intellij.openapi.diagnostic.RuntimeExceptionWithAttachments: com.intellij.diagnostic.PluginException: annotator: TemplateLintExternalAnnotator@17c3fc95 (class TemplateLintExternalAnnotator) [Plugin: com.emberjs.experimental]
	at com.intellij.idea.IdeaLogger.error(IdeaLogger.java:111)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.processError(ExternalToolPass.java:278)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.doAnnotate(ExternalToolPass.java:234)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.doAnnotate(ExternalToolPass.java:219)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass$1.lambda$run$0(ExternalToolPass.java:177)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.runChangeAware(ExternalToolPass.java:292)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass$1.lambda$run$2(ExternalToolPass.java:177)
	at com.intellij.openapi.progress.util.BackgroundTaskUtil.lambda$runUnderDisposeAwareIndicator$15(BackgroundTaskUtil.java:371)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$1(CoreProgressManager.java:229)
	at com.intellij.platform.diagnostic.telemetry.helpers.TraceKt.use(trace.kt:43)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:228)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$14(CoreProgressManager.java:680)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:755)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:711)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:679)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:77)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:209)
	at com.intellij.openapi.progress.util.BackgroundTaskUtil.runUnderDisposeAwareIndicator(BackgroundTaskUtil.java:366)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass$1.run(ExternalToolPass.java:175)
	at com.intellij.util.ui.update.Update.execute$suspendImpl(Update.kt:60)
	at com.intellij.util.ui.update.Update.execute(Update.kt)
	at com.intellij.util.ui.update.MergingUpdateQueue.executeUpdates$suspendImpl(MergingUpdateQueue.kt:364)
	at com.intellij.util.ui.update.MergingUpdateQueue.executeUpdates(MergingUpdateQueue.kt)
	at com.intellij.util.ui.update.MergingUpdateQueue$restart$1.invokeSuspend(MergingUpdateQueue.kt:344)
	at com.intellij.util.ui.update.MergingUpdateQueue$restart$1.invoke(MergingUpdateQueue.kt)
	at com.intellij.util.ui.update.MergingUpdateQueue$restart$1.invoke(MergingUpdateQueue.kt)
	at com.intellij.util.SingleAlarm$scheduleTask$1$1$1.invokeSuspend(SingleAlarm.kt:394)
	at com.intellij.util.SingleAlarm$scheduleTask$1$1$1.invoke(SingleAlarm.kt)
	at com.intellij.util.SingleAlarm$scheduleTask$1$1$1.invoke(SingleAlarm.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:62)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:155)
	at kotlinx.coroutines.BuildersKt.withContext(Unknown Source)
	at com.intellij.util.SingleAlarm$scheduleTask$1$1.invokeSuspend(SingleAlarm.kt:392)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:608)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:873)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:763)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:750)
Caused by: com.intellij.diagnostic.PluginException: annotator: TemplateLintExternalAnnotator@17c3fc95 (class TemplateLintExternalAnnotator) [Plugin: com.emberjs.experimental]
	at com.intellij.diagnostic.PluginProblemReporterImpl.createPluginExceptionByClass(PluginProblemReporterImpl.java:23)
	at com.intellij.diagnostic.PluginException.createByClass(PluginException.java:90)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.processError(ExternalToolPass.java:277)
	... 37 more
Caused by: java.lang.NoClassDefFoundError: org/codehaus/jettison/json/JSONObject
	at TemplateLintResultParser.parse(TemplateLintResultParser.kt:46)
	at TemplateLintExternalRunner$Companion.runProcess(TemplateLintExternalRunner.kt:60)
	at TemplateLintExternalRunner$Companion.templateLint(TemplateLintExternalRunner.kt:34)
	at TemplateLintExternalRunner$Companion.access$templateLint(TemplateLintExternalRunner.kt:28)
	at TemplateLintExternalRunner.execute(TemplateLintExternalRunner.kt:222)
	at TemplateLintExternalRunner.highlight(TemplateLintExternalRunner.kt:179)
	at TemplateLintExternalAnnotator.annotate(TemplateLintExternalAnnotator.kt:50)
	at TemplateLintExternalAnnotator.annotate(TemplateLintExternalAnnotator.kt:17)
	at com.intellij.lang.javascript.linter.JSLinterWithInspectionExternalAnnotator.doAnnotate(JSLinterWithInspectionExternalAnnotator.java:139)
	at com.intellij.lang.javascript.linter.JSLinterWithInspectionExternalAnnotator.doAnnotate(JSLinterWithInspectionExternalAnnotator.java:22)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.lambda$doAnnotate$1(ExternalToolPass.java:227)
	at com.intellij.codeInsight.daemon.impl.AnnotationSessionImpl.computeWithSession(AnnotationSessionImpl.java:87)
	at com.intellij.codeInsight.daemon.impl.AnnotationSessionImpl.computeWithSession(AnnotationSessionImpl.java:79)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.doAnnotate(ExternalToolPass.java:225)
	... 36 more
Caused by: java.lang.ClassNotFoundException: org.codehaus.jettison.json.JSONObject 
```

Not sure if this is helpful at all.

--- a/src/main/kotlin/com/emberjs/glint/GlintLspSupportProvider.kt
+++ b/src/main/kotlin/com/emberjs/glint/GlintLspSupportProvider.kt
@@ -93,6 +93,7 @@ class GlintLspServerDescriptor(private val myProject: Project) : LspServerDescri
                 f = f.parent
             }
             if (f?.path?.contains("node_modules") == true) {
+                f = f.parentEmberModule
                 continue
             }
             if (f != null && isAvailableFromDir(f)) {

--- a/src/main/kotlin/com/emberjs/hbs/linter/ember-template-lint/TemplateLintResultParser.kt
+++ b/src/main/kotlin/com/emberjs/hbs/linter/ember-template-lint/TemplateLintResultParser.kt
@@ -2,7 +2,7 @@ import com.intellij.lang.annotation.HighlightSeverity
 import com.intellij.lang.javascript.linter.JSLinterError
 import com.intellij.openapi.diagnostic.Logger
 import com.intellij.openapi.util.text.StringUtil
-import org.codehaus.jettison.json.JSONObject
+import kotlinx.serialization.json.*
 import java.io.IOException
 import java.util.*
 
@@ -19,13 +19,13 @@ class TemplateLintResultParser {
         private const val SEVERITY = "severity"
         private const val MESSAGE = "message"
 
-        private fun parseItem(map: JSONObject): JSLinterError {
-            val line = map.getInt(LINE)
-            val column = map.getInt(COLUMN)
-            val rule = map.getString(RULE)
-            val text = map.getString(MESSAGE)
+        private fun parseItem(map: JsonObject): JSLinterError {
+            val line = map[LINE]!!.jsonPrimitive.int
+            val column = map[COLUMN]!!.jsonPrimitive.int
+            val rule = map[RULE]!!.jsonPrimitive.content
+            val text = map[MESSAGE]!!.jsonPrimitive.content
 
-            val highlightSeverity = when (map.getInt(SEVERITY)) {
+            val highlightSeverity = when (map[SEVERITY]!!.jsonPrimitive.int) {
                 WARNING_SEVERITY -> HighlightSeverity.WARNING
                 ERROR_SEVERITY -> HighlightSeverity.ERROR
                 else -> null
@@ -43,14 +43,14 @@ class TemplateLintResultParser {
 
         val errorList: ArrayList<JSLinterError> = ArrayList()
         try {
-            val obj = JSONObject(stdout)
+            val obj = Json.parseToJsonElement(stdout!!).jsonObject
 
-            val issues = obj.getJSONArray(obj.keys().next() as String)
-            for (i in 0 until issues.length()) {
-                val issue = issues.getJSONObject(i)
+            val issues = obj.jsonArray
+            for (i in 0 until issues.size) {
+                val issue = issues[i].jsonObject
 
                 // we skip fatal errors
-                if (issue.has("fatal") && issue.getBoolean("fatal")) continue
+                if (issue.containsKey("fatal") && issue["fatal"]?.jsonPrimitive?.boolean == true) continue
 
                 errorList.add(parseItem(issue))
             }

--- a/src/main/kotlin/com/emberjs/hbs/linter/ember-template-lint/TemplateLintResultParser.kt
+++ b/src/main/kotlin/com/emberjs/hbs/linter/ember-template-lint/TemplateLintResultParser.kt
@@ -45,7 +45,7 @@ class TemplateLintResultParser {
         try {
             val obj = Json.parseToJsonElement(stdout!!).jsonObject
 
-            val issues = obj.jsonArray
+            val issues = obj.values.flatMap { map -> map.jsonArray }
             for (i in 0 until issues.size) {
                 val issue = issues[i].jsonObject
 
@@ -59,7 +59,7 @@ class TemplateLintResultParser {
         } catch (ioException: IOException) {
             return null
         } catch (exception: Exception) {
-            LOG.warn("TemplateLint result parsing error")
+            LOG.warn("TemplateLint result parsing error: $exception")
             throw Exception("TemplateLint result parsing error", exception)
         }
     }

